bug: The test case does not take into account tengine

### Issue description
When I tried to migrate CI from Travis to GitHub action, I found that this test case does not pass when using Tengine.
https://github.com/wingsjj/incubator-apisix/runs/824980221?check_suite_focus=true#step:6:130

### Environment

* apisix version (cmd: `apisix version`): 1.4
* OS: Ubuntu 18.04

### Minimal test code / Steps to reproduce the issue
1.https://github.com/wingsjj/incubator-apisix/runs/824980221?check_suite_focus=true#step:6:130
2.https://github.com/apache/incubator-apisix/blob/master/t/node/remote-addr-ipv6.t#L117

### What's the actual result? (including assertion message & call stack if applicable)
> Failed test 'TEST 4: IPv6 /hello - response_body - response is expected (repeated req 0, req 0)'
> at /home/runner/work/incubator-apisix/incubator-apisix/test-nginx/lib/Test/Nginx/Socket.pm line 1589.
> @@ -3,7 +3,7 @@
> received: HTTP/1.1 200 OK
>  received: Content-Type: text/plain
>  received: Connection: close
> -received: Server: openresty
> +received: Server: Tengine
>  received: 
>  received: hello world
>  failed to receive a line: closed [] 

> Test Summary Report
> -------------------
> t/node/remote-addr-ipv6.t            (Wstat: 256 Tests: 15 Failed: 1)
>   Failed test:  11
>   Non-zero exit status: 1
> Files=119, Tests=3969, 959 wallclock secs ( 0.95 usr  0.18 sys + 130.74 cusr 27.03 csys = 158.90 CPU)
> Result: FAIL
> ##[error]Process completed with exit code 1.


### What's the expected result?
Test case passed successfully.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -19,6 +19,10 @@ jobs:
     runs-on: ${{ matrix.platform }}
 
     steps:
+      - name: Test ipv6
+        run: |
+          ifconfig
+          ping6 google.com
         
       - name: Check out code
         uses: actions/checkout@v2

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -19,6 +19,11 @@ jobs:
     runs-on: ${{ matrix.platform }}
 
     steps:
+      - name: Test ipv6
+        run: |
+          sudo apt install -y curl
+          curl -6 www.google.com
+        
       - name: Check out code
         uses: actions/checkout@v2
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -22,7 +22,7 @@ jobs:
       - name: Check out code
         uses: actions/checkout@v2
 
-      - name: ${{ matrix.platform }} - Get dependencies
+      - name: Get dependencies - ${{ matrix.platform }}
         if: matrix.platform == 'ubuntu-18.04'
         run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
       
@@ -31,7 +31,6 @@ jobs:
         run: |
           sudo chown -R `whoami` /usr/local/Homebrew/
           sudo chown -R $(whoami) $(brew --prefix)/*
-          sudo mkdir /usr/local/Frameworks
           sudo chown -R `whoami` /usr/local/Frameworks/
 
       - name: Before install

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -16,7 +16,7 @@ jobs:
             os_name: osx_openresty
       
 
-    runs-on: ${{ matrix.os }}
+    runs-on: ${{ matrix.platform }}
 
     steps:
       - name: Check out code
@@ -26,13 +26,13 @@ jobs:
         run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
 
       - name: Before install
-        run: ./.travis/${{ matrix.os.os_name }}_runner.sh before_install
+        run: ./.travis/${{ matrix.os_name }}_runner.sh before_install
 
       - name: Install
-        run: ./.travis/${{ matrix.os.os_name }}_runner.sh do_install
+        run: ./.travis/${{ matrix.os_name }}_runner.sh do_install
 
       - name: Script
-        run: ./.travis/${{ matrix.os.os_name }}_runner.sh script
+        run: ./.travis/${{ matrix.os_name }}_runner.sh script
 
       - name: After success
-        run: ./.travis/${{ matrix.os.os_name }}_runner.sh after_success
+        run: ./.travis/${{ matrix.os_name }}_runner.sh after_success

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -2,8 +2,9 @@ name: CI
 
 on:
   push:
-    branches:
-      - master
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
 
 jobs:
   build:

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -34,9 +34,6 @@ jobs:
       - name: Linux Install
         if: matrix.platform == 'ubuntu-18.04'
         run: sudo ./.travis/${{ matrix.os_name }}_runner.sh do_install
-      
-      - name: Apisix version
-        run: apisix version
 
       - name: Linux Script
         if: matrix.platform == 'ubuntu-18.04'

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -18,12 +18,7 @@ jobs:
 
     runs-on: ${{ matrix.platform }}
 
-    steps:
-      - name: Test ipv6
-        run: |
-          ifconfig
-          ping6 google.com
-        
+    steps:        
       - name: Check out code
         uses: actions/checkout@v2
 
@@ -39,6 +34,9 @@ jobs:
       - name: Linux Install
         if: matrix.platform == 'ubuntu-18.04'
         run: sudo ./.travis/${{ matrix.os_name }}_runner.sh do_install
+      
+      - name: Apisix version
+        run: apisix version
 
       - name: Linux Script
         if: matrix.platform == 'ubuntu-18.04'

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -19,10 +19,6 @@ jobs:
     runs-on: ${{ matrix.platform }}
 
     steps:
-      - name: Test ipv6
-        run: |
-          sudo apt install -y curl
-          curl -6 www.google.com
         
       - name: Check out code
         uses: actions/checkout@v2

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -8,13 +8,13 @@ on:
 jobs:
   build:
     strategy:
+      fail-fast: false
       matrix:
         platform: [ubuntu-18.04]
         os_name: [linux_openresty, linux_tengine, linux_apisix_master_luarocks, linux_apisix_current_luarocks]
         include:
           - platform: macos-latest
             os_name: osx_openresty
-      
 
     runs-on: ${{ matrix.platform }}
 
@@ -22,8 +22,13 @@ jobs:
       - name: Check out code
         uses: actions/checkout@v2
 
-      - name: Get dependencies
+      - name: ${{ matrix.platform }} - Get dependencies
+        if: matrix.platform == ubuntu-18.04
         run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
+        
+      - name: ${{ matrix.platform }} - Get dependencies
+        if: matrix.platform == macos-latest
+        run: brew install cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
 
       - name: Before install
         run: ./.travis/${{ matrix.os_name }}_runner.sh before_install

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -22,25 +22,39 @@ jobs:
       - name: Check out code
         uses: actions/checkout@v2
 
+#----------------------------------------------------------------------------
       - name: Get dependencies - ${{ matrix.platform }}
         if: matrix.platform == 'ubuntu-18.04'
         run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
-      
-      - name: Preprocessing for macos
-        if: matrix.platform == 'macos-latest'
-        run: |
-          sudo chown -R `whoami` /usr/local/Homebrew/
-          sudo chown -R $(whoami) $(brew --prefix)/*
-          sudo chown -R `whoami` /usr/local/Frameworks/
 
-      - name: Before install
+      - name: Linux Before install
+        if: matrix.platform == 'ubuntu-18.04'
         run: sudo ./.travis/${{ matrix.os_name }}_runner.sh before_install
 
-      - name: Install
+      - name: Linux Install
+        if: matrix.platform == 'ubuntu-18.04'
         run: sudo ./.travis/${{ matrix.os_name }}_runner.sh do_install
 
-      - name: Script
+      - name: Linux Script
+        if: matrix.platform == 'ubuntu-18.04'
         run: sudo ./.travis/${{ matrix.os_name }}_runner.sh script
 
-      - name: After success
+      - name: Linux After success
+        if: matrix.platform == 'ubuntu-18.04'
         run: sudo ./.travis/${{ matrix.os_name }}_runner.sh after_success
+#----------------------------------------------------------------------------
+      - name: MacOS Before install
+        if: matrix.platform == 'macos-latest'
+        run: ./.travis/${{ matrix.os_name }}_runner.sh before_install
+
+      - name: MacOS Install
+        if: matrix.platform == 'macos-latest'
+        run: ./.travis/${{ matrix.os_name }}_runner.sh do_install
+
+      - name: MacOS Script
+        if: matrix.platform == 'macos-latest'
+        run: ./.travis/${{ matrix.os_name }}_runner.sh script
+
+      - name: MacOS After success
+        if: matrix.platform == 'macos-latest'
+        run: ./.travis/${{ matrix.os_name }}_runner.sh after_success

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -23,11 +23,11 @@ jobs:
         uses: actions/checkout@v2
 
       - name: ${{ matrix.platform }} - Get dependencies
-        if: matrix.platform == ubuntu-18.04
+        if: matrix.platform == 'ubuntu-18.04'
         run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
         
       - name: ${{ matrix.platform }} - Get dependencies
-        if: matrix.platform == macos-latest
+        if: matrix.platform == 'macos-latest'
         run: brew install cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
 
       - name: Before install

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -23,7 +23,7 @@ jobs:
         uses: actions/checkout@v2
 
 #----------------------------------------------------------------------------
-      - name: Get dependencies - ${{ matrix.platform }}
+      - name: Linux Get dependencies
         if: matrix.platform == 'ubuntu-18.04'
         run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -1,0 +1,38 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        platform: [ubuntu-18.04]
+        os_name: [linux_openresty, linux_tengine, linux_apisix_master_luarocks, linux_apisix_current_luarocks]
+        include:
+          - platform: macos-latest
+            os_name: osx_openresty
+      
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Get dependencies
+        run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
+
+      - name: Before install
+        run: ./.travis/${{ matrix.os.os_name }}_runner.sh before_install
+
+      - name: Install
+        run: ./.travis/${{ matrix.os.os_name }}_runner.sh do_install
+
+      - name: Script
+        run: ./.travis/${{ matrix.os.os_name }}_runner.sh script
+
+      - name: After success
+        run: ./.travis/${{ matrix.os.os_name }}_runner.sh after_success

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -17,6 +17,9 @@ jobs:
             os_name: osx_openresty
 
     runs-on: ${{ matrix.platform }}
+    env:
+      SERVER_NAME: ${{ matrix.os_name }}
+
 
     steps:        
       - name: Check out code

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -25,10 +25,6 @@ jobs:
       - name: ${{ matrix.platform }} - Get dependencies
         if: matrix.platform == 'ubuntu-18.04'
         run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
-        
-      - name: ${{ matrix.platform }} - Get dependencies
-        if: matrix.platform == 'macos-latest'
-        run: brew install cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
 
       - name: Before install
         run: ./.travis/${{ matrix.os_name }}_runner.sh before_install

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -39,9 +39,9 @@ jobs:
         if: matrix.platform == 'ubuntu-18.04'
         run: sudo ./.travis/${{ matrix.os_name }}_runner.sh script
 
-      - name: Linux After success
-        if: matrix.platform == 'ubuntu-18.04'
-        run: sudo ./.travis/${{ matrix.os_name }}_runner.sh after_success
+#       - name: Linux After success
+#         if: matrix.platform == 'ubuntu-18.04'
+#         run: sudo ./.travis/${{ matrix.os_name }}_runner.sh after_success
 #----------------------------------------------------------------------------
       - name: MacOS Before install
         if: matrix.platform == 'macos-latest'
@@ -55,6 +55,6 @@ jobs:
         if: matrix.platform == 'macos-latest'
         run: ./.travis/${{ matrix.os_name }}_runner.sh script
 
-      - name: MacOS After success
-        if: matrix.platform == 'macos-latest'
-        run: ./.travis/${{ matrix.os_name }}_runner.sh after_success
+#       - name: MacOS After success
+#         if: matrix.platform == 'macos-latest'
+#         run: ./.travis/${{ matrix.os_name }}_runner.sh after_success

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -25,6 +25,14 @@ jobs:
       - name: ${{ matrix.platform }} - Get dependencies
         if: matrix.platform == 'ubuntu-18.04'
         run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
+      
+      - name: Preprocessing for macos
+        if: matrix.platform == 'macos-latest'
+        run: |
+          sudo chown -R `whoami` /usr/local/Homebrew/
+          sudo chown -R $(whoami) $(brew --prefix)/*
+          sudo mkdir /usr/local/Frameworks
+          sudo chown -R `whoami` /usr/local/Frameworks/
 
       - name: Before install
         run: sudo ./.travis/${{ matrix.os_name }}_runner.sh before_install

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -27,13 +27,13 @@ jobs:
         run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl etcd
 
       - name: Before install
-        run: ./.travis/${{ matrix.os_name }}_runner.sh before_install
+        run: sudo ./.travis/${{ matrix.os_name }}_runner.sh before_install
 
       - name: Install
-        run: ./.travis/${{ matrix.os_name }}_runner.sh do_install
+        run: sudo ./.travis/${{ matrix.os_name }}_runner.sh do_install
 
       - name: Script
-        run: ./.travis/${{ matrix.os_name }}_runner.sh script
+        run: sudo ./.travis/${{ matrix.os_name }}_runner.sh script
 
       - name: After success
-        run: ./.travis/${{ matrix.os_name }}_runner.sh after_success
+        run: sudo ./.travis/${{ matrix.os_name }}_runner.sh after_success

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,6 +2,8 @@ dist: bionic
 sudo: required
 
 matrix:
+  allow_failures:
+    - os: osx
   include:
     - os: linux
       services:

--- a/.travis/linux_openresty_runner.sh
+++ b/.travis/linux_openresty_runner.sh
@@ -49,7 +49,7 @@ before_install() {
     docker pull bitinit/eureka
     docker run --name eureka -d -p 8761:8761 --env ENVIRONMENT=apisix --env spring.application.name=apisix-eureka --env server.port=8761 --env eureka.instance.ip-address=127.0.0.1 --env eureka.client.registerWithEureka=true --env eureka.client.fetchRegistry=false --env eureka.client.serviceUrl.defaultZone=http://127.0.0.1:8761/eureka/ bitinit/eureka
     sleep 5
-    docker exec -it kafka-server1 /opt/bitnami/kafka/bin/kafka-topics.sh --create --zookeeper zookeeper-server:2181 --replication-factor 1 --partitions 1 --topic test2
+    docker exec -i kafka-server1 /opt/bitnami/kafka/bin/kafka-topics.sh --create --zookeeper zookeeper-server:2181 --replication-factor 1 --partitions 1 --topic test2
 }
 
 do_install() {

--- a/.travis/linux_tengine_runner.sh
+++ b/.travis/linux_tengine_runner.sh
@@ -50,7 +50,7 @@ before_install() {
     docker pull bitinit/eureka
     docker run --name eureka -d -p 8761:8761 --env ENVIRONMENT=apisix --env spring.application.name=apisix-eureka --env server.port=8761 --env eureka.instance.ip-address=127.0.0.1 --env eureka.client.registerWithEureka=true --env eureka.client.fetchRegistry=false --env eureka.client.serviceUrl.defaultZone=http://127.0.0.1:8761/eureka/ bitinit/eureka
     sleep 5
-    docker exec -it kafka-server1 /opt/bitnami/kafka/bin/kafka-topics.sh --create --zookeeper zookeeper-server:2181 --replication-factor 1 --partitions 1 --topic test2
+    docker exec -i kafka-server1 /opt/bitnami/kafka/bin/kafka-topics.sh --create --zookeeper zookeeper-server:2181 --replication-factor 1 --partitions 1 --topic test2
 }
 
 tengine_install() {

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -120,7 +120,7 @@ request sent: 59
 received: HTTP/1.1 200 OK
 received: Content-Type: text/plain
 received: Connection: close
-received: Server: $::server
+received: Server: "$::server"
 received: 
 received: hello world
 failed to receive a line: closed []

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -26,6 +26,12 @@ if ((defined $travis_os_name) && $travis_os_name eq "linux") {
     plan 'no_plan';
 }
 
+my $server_name = $ENV{SERVER_NAME};
+our $server = openresty;
+if ((defined $server_name) && $server_name eq "linux_tengine") {
+    $server = Tengine;
+}
+
 run_tests();
 
 __DATA__
@@ -114,7 +120,7 @@ request sent: 59
 received: HTTP/1.1 200 OK
 received: Content-Type: text/plain
 received: Connection: close
-received: Server: Tengine
+received: Server: $::server
 received: 
 received: hello world
 failed to receive a line: closed []

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -115,7 +115,7 @@ location /t {
 --- request
 GET /t
 --- response_body eval
-connected: 1
+qr{connected: 1
 request sent: 59
 received: HTTP/1.1 200 OK
 received: Content-Type: text/plain
@@ -124,7 +124,7 @@ received: Server: "$::server"
 received: 
 received: hello world
 failed to receive a line: closed []
-close: 1 nil
+close: 1 nil}
 --- no_error_log
 [error]
 

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -114,7 +114,7 @@ location /t {
 }
 --- request
 GET /t
---- response_body
+--- response_body eval
 connected: 1
 request sent: 59
 received: HTTP/1.1 200 OK

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -114,7 +114,7 @@ request sent: 59
 received: HTTP/1.1 200 OK
 received: Content-Type: text/plain
 received: Connection: close
-received: Server: openresty
+received: Server: Tengine
 received: 
 received: hello world
 failed to receive a line: closed []

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -109,8 +109,7 @@ location /t {
 --- request
 GET /t
 --- response_body eval
-qr{
-connected: 1
+qr{connected: 1
 request sent: 59
 received: HTTP/1.1 200 OK
 received: Content-Type: text/plain
@@ -119,8 +118,7 @@ received: Server: \w+
 received: 
 received: hello world
 failed to receive a line: closed \[\]
-close: 1 nil
-}
+close: 1 nil}
 --- no_error_log
 [error]
 

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -26,12 +26,6 @@ if ((defined $travis_os_name) && $travis_os_name eq "linux") {
     plan 'no_plan';
 }
 
-my $server_name = $ENV{SERVER_NAME};
-our $server = "openresty";
-if ((defined $server_name) && $server_name eq "linux_tengine") {
-    $server = "Tengine";
-}
-
 run_tests();
 
 __DATA__

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -120,9 +120,8 @@ request sent: 59
 received: HTTP/1.1 200 OK
 received: Content-Type: text/plain
 received: Connection: close
-received: Server: "$::server"
-received: 
-received: hello world
+received: Server: \w+
+received: \nreceived: hello world
 failed to receive a line: closed []
 close: 1 nil}
 --- no_error_log

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -27,9 +27,9 @@ if ((defined $travis_os_name) && $travis_os_name eq "linux") {
 }
 
 my $server_name = $ENV{SERVER_NAME};
-our $server = openresty;
+our $server = "openresty";
 if ((defined $server_name) && $server_name eq "linux_tengine") {
-    $server = Tengine;
+    $server = "Tengine";
 }
 
 run_tests();

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -115,7 +115,8 @@ received: HTTP/1.1 200 OK
 received: Content-Type: text/plain
 received: Connection: close
 received: Server: \w+
-received: \nreceived: hello world
+received: 
+received: hello world
 failed to receive a line: closed []
 close: 1 nil}
 --- no_error_log

--- a/t/node/remote-addr-ipv6.t
+++ b/t/node/remote-addr-ipv6.t
@@ -109,7 +109,8 @@ location /t {
 --- request
 GET /t
 --- response_body eval
-qr{connected: 1
+qr{
+connected: 1
 request sent: 59
 received: HTTP/1.1 200 OK
 received: Content-Type: text/plain
@@ -117,8 +118,9 @@ received: Connection: close
 received: Server: \w+
 received: 
 received: hello world
-failed to receive a line: closed []
-close: 1 nil}
+failed to receive a line: closed \[\]
+close: 1 nil
+}
 --- no_error_log
 [error]
 

